web/src/engine/predictive-text/worker-thread/src/main/transform-subsets.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ *
+ * Created by jahorton on 2025-08-21
+ *
+ * This file defines a method used to 'catalog' transforms based on the area of
+ * the context that they impact, allowing groups of similar transforms to be
+ * handled as related groups more easily.
+ */
+
+import { LexicalModelTypes } from '@keymanapp/common-types';
+import { KMWString } from '@keymanapp/web-utils';
+
+import Distribution = LexicalModelTypes.Distribution;
+import Transform = LexicalModelTypes.Transform;
+
+/**
+ * Represents a subset of Transforms, taken from the distribution of input
+ * Transform, where all members meet certain criteria that others from the
+ * source distribution do not.
+ * @typeParam Key - the type of the criteria uniquely keying this Transform
+ * subset.
+ */
+export interface TransformSubset<Key> {
+  /**
+   * The value of whatever property was used as a criteria when building this
+   * distinct subset.
+   */
+  key: Key;
+
+  /**
+   * All Transforms meeting the criteria defining this TransformSubset
+   */
+  entries: Distribution<Transform>;
+
+  /**
+   * The summed probability mass for all Transforms in the subset
+   */
+  cumulativeMass: number;
+
+  /**
+   * The currently-processed and commonly-held portion of the `insert` field
+   * for all Transforms in this subset.
+   */
+  insert: string
```

Comment:
Is `insert` used anywhere? At least it's not checked in the tests, and when we create a `TransformSubset` it always gets initialized with the empty string.

Reply:
See https://github.com/keymanapp/keyman/pull/14592#discussion_r2301112371.